src/components/Editor/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { AllHTMLAttributes } from 'react';
+import Editor from 'react-simple-code-editor';
+// @ts-expect-error type doesn't exist
+import { highlight, languages } from 'prismjs/components/prism-core';
+import 'prismjs/components/prism-graphql';
+import 'prismjs/themes/prism.css';
+import EditorContextMenu, { type EditorContextMenuItems } from './contextMenu';
+
+export { languages };
+
+const hightlightWithLineNumbers = (
+  input: string,
+  language = languages.graphql
+) =>
+  (highlight(input, language) as string)
+    .split('\n')
+    .map(
+      (line, i) =>
+        `<span class='absolute p-1 font-bold text-xs -left-4 w-12 text-right dark:bg-blue-500 bg-slate-500'>${
```

Comment:
Look, there is a cool trick for line printing using CSS only. Looks good, sounds even better!

https://stackoverflow.com/questions/40842277/create-line-numbers-on-pre-with-css-only

Reply:
Great CSS property, however, It won't make much difference, since I will need to prepend span to each line of textarea.value. But it would work great with JSON viewer.